src/instrumenter/annotations.ts

```diff
@@ -676,36 +659,81 @@ function processMacroAnnotations(annotationMap: AnnotationMap, compilerVersion:
                     }
                 };
 
-                /**
-                 * @todo Using `inclusive = true` here seems to be a design oversight.
-                 * Better to fix this in **solc-typed-ast**.
-                 */
-                const targets = resolveAny(name, scope, compilerVersion, true);
-                const target = [...targets][0];
+                const targets = [...resolveAny(name, scope, ctx.compilerVersion, true)];
+                const target = single(
+                    targets,
+                    'Unable to pick single target entity for name "{0}" in scope of {1}',
+                    name,
+                    scope
+                );
 
                 if (target instanceof FunctionDefinition && args.length > 0) {
                     const params = target.vParameters.vParameters;
+                    const pairs = zip(
+                        args,
+                        params,
+                        `{0}: arguments count {1} in macro definition exceeds method arguments count {2}`,
```

Comment:
The error message here doesn't match the behavior of `zip`. `zip` will fail if the lengths are not exactly equal.

Reply:
Addressed in https://github.com/ConsenSys/scribble/pull/96/commits/24012c080b685c49cbcc930f86a107cfee582a84